llpc/patch/llpcPatchDescriptorLoad.cpp

```diff
@@ -509,8 +527,35 @@ Value* PatchDescriptorLoad::LoadDescriptor(
         }
         else
         {
-            auto pDescOffset = ConstantInt::get(Type::getInt32Ty(*m_pContext), descOffset);
-            auto pDescSize   = ConstantInt::get(Type::getInt32Ty(*m_pContext), descSize, 0);
+            llvm::Value* pDescOffset = nullptr;
+            if (!m_pPipelineState->GetBuilderContext()->BuildingRelocatableElf())
+            {
+                pDescOffset = ConstantInt::get(Type::getInt32Ty(*m_pContext), descOffset);
+            }
+            else
+            {
+                IRBuilder<> builder(*m_pContext);
+                builder.SetInsertPoint(pInsertPoint);
+                std::stringstream ssRelocName;
+                ssRelocName << "doff_" << descSet << "_" << binding;
+                if (nodeType1 == ResourceMappingNodeType::DescriptorSampler)
+                {
+                    ssRelocName << "_s";
+                }
+                else if (nodeType1 == ResourceMappingNodeType::DescriptorResource)
+                {
+                    ssRelocName << "_r";
+                }
+                else
+                {
+                    ssRelocName << "_b";
+                }
+                pDescOffset = builder.CreateIntrinsic(Intrinsic::amdgcn_reloc_constant, {}, {
+                    llvm::MetadataAsValue::get(*m_pContext, llvm::MDNode::get(*m_pContext,
+                    llvm::MDString::get(*m_pContext, ssRelocName.str().c_str())))
```

Comment:
I don't know what the style guide says.  This is hard to read because it has two lines forming one statement, but both are at the same level of indentation.

Reply:
Reformatted to the same line.